os/migrating-to-clcs.md

```diff
@@ -0,0 +1,377 @@
+```
+
+This section could also be used to define systemd drop-in files for existing units.
+
+```cloud-config
+#cloud-config
+
+coreos:
+  units:
+    - name: "docker.service"
+      drop-ins:
+        - name: "50-insecure-registry.conf"
+          content: |
+            [Service]
+            Environment=DOCKER_OPTS='--insecure-registry="10.0.1.0/24"'
+```
+
+And existing units could also be started without any further configuration.
+
+```cloud-config
+#cloud-config
+
+coreos:
+  units:
+    - name: "etcd2.service"
+      command: "start"
+```
+
+One big difference in Container Linux Config compared to cloud configs is that the configuration is applied via [Ignition][ignition] before the machine has fully booted, as opposed to coreos-cloudinit that runs after the machine has fully booted. As a result units cannot be directly started in a Container Linux Config, the unit is instead enabled so that systemd will begin the unit once systemd starts.
+
+_Note: in this example an `[Install]` section has been added so that the unit can be enabled._
+
+```container-linux-config
+systemd:
+  units:
+    - name: "docker-redis.service"
+      enable: true
+      contents: |
+        [Unit]
+        Description=Redis container
+        Author=Me
+        After=docker.service
+
+        [Service]
+        Restart=always
+        ExecStart=/usr/bin/docker start -a redis_server
+        ExecStop=/usr/bin/docker stop -t 2 redis_server
+
+        [Install]
+        WantedBy=multi-user.target
+```
+
+Drop-in files can be provided for units in a Container Linux Config just like in a cloud config.
+
+```container-linux-config
+systemd:
+  units:
+    - name: "docker.service"
+      dropins:
+        - name: "50-insecure-registry.conf"
+          contents: |
+            [Service]
+            Environment=DOCKER_OPTS='--insecure-registry="10.0.1.0/24"'
+```
+
+Existing units can also be enabled without configuration.
+
+```container-linux-config
+systemd:
+  units:
+    - name: "etcd-member.service"
+      enable: true
+```
+
+### ssh_authorized_keys
+
+In a cloud config the `ssh_authorized_keys` section can be used to add ssh public keys to the `core` user.
+
+```cloud-config
+#cloud-config
+
+ssh_authorized_keys:
+  - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0g+ZTxC7weoIJLUafOgrm+h..."
+```
+
+In a Container Linux Config there is no analogous section to `ssh_authorized_keys`, but ssh keys for the core user can be set just as easily using the `passwd.users.*` section:
+
+```container-linux-config
+passwd:
+  users:
+    - name: core
+      ssh_authorized_keys:
+        - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0g+ZTxC7weoIJLUafOgrm+h..."
+```
+
+### hostname
+
+In a cloud config the `hostname` section can be used to set a machine's hostname.
+
+```cloud-config
+#cloud-config
+
+hostname: "coreos1"
+```
+
+There is no analogous section in a Container Linux Config, so to set the hostname the `storage.files.*` section can be used to write `/etc/hostname`.
```

Comment:
Eh, we could very easily add syntactic sugar for setting the hostname (maybe we should).